File: app/channels/chat_channel.rb
class ChatChannel < ApplicationCable::Channel
  def subscribed
    # stream_from "some_channel"
    stream_from "chat_channel"
  end

  def unsubscribed
    # Any cleanup needed when channel is unsubscribed
  end

  def send_message
    @message = Message.new(body: data['message'])
    if @message.save
      ActionCable.server.broadcast "chat_channel", message: @message.body
    end
  end
end

File: app/javascript/channels/chat_channel.js
import consumer from "channels/consumer"

var createObject = (object) => {
  var template = document.querySelector('template');

  var node = document.importNode(template.content, true);

  node.querySelector('div').className += ` msg-${object.pc}`;

  node.querySelector('div div small').innerText = object.owner;
  node.querySelector('div div span').innerText = object.message;
  node.querySelector('div div span').className += ` baloon-${object.pc}`;
  node.querySelectorAll('div div small')[1].innerText = object.time;

  var chatbox = document.querySelector('.chatbox');

  chatbox.appendChild(node)

  chatbox.scrollTop = chatbox.scrollHeight
}

consumer.subscriptions.create("ChatChannel", {
  connected() {
    // Called when the subscription is ready for use on the server
  },

  disconnected() {
    // Called when the subscription has been terminated by the server
  },

  received(data) {
    // Called when there's incoming data on the websocket for this channel
    createObject(data.object)
  }
});
